addons/hr_contract/models/resource_resource.py

```diff
@@ -15,7 +15,23 @@ def _get_calendars_validity_within_period(self, start, end, default_company=None
         if not self:
             return super()._get_calendars_validity_within_period(start, end, default_company=default_company)
         calendars_within_period_per_resource = defaultdict(lambda: defaultdict(Intervals))  # keys are [resource id:integer][calendar:self.env['resource.calendar']]
-        resource_without_contract = self.filtered(lambda r: not r.employee_id or r.employee_id.employee_type not in ['employee', 'student'])
+        employee_ids_with_active_contracts = {
+            contract['employee_id'][0] for contract in
+            self.env['hr.contract']._read_group(
+                domain=[
+                    ('employee_id', 'in', self.employee_id.ids),
+                    '|', ('state', '=', 'open'),
+                    '|', ('state', '=', 'close'),
+                         '&', ('state', '=', 'draft'), ('kanban_state', '=', 'done')
```

Comment:
Is the contract really active if it's expired or in draft?

Reply:
'employee_ids_with_active_contracts' stands for 'employee_ids_that_have_or_had_an_active_contract'.
Thought it was a bit lighter :sweat_smile:

Reply:
I see, thanks, maybe you could add a comment then, because it's confusing as is